pkgs/os-specific/linux/fxload/default.nix

```diff
@@ -1,37 +1,30 @@
-{lib, stdenv, fetchurl}:
+{ lib
+, stdenv
+, libusb1
+}:
 
 stdenv.mkDerivation rec {
   pname = "fxload";
-  version = "2002.04.11";
+  version = libusb1.version;
+  libusb = libusb1.override { withExamples = true; };
```

Comment:
Should this `libusb` line be in `passthru` rather than a top-level `mkDerivation` attr? (e.g. `passthru.libusb = libusb1.override { ... }` and use it later on as `passthru.libusb`)

Reply:
passthru does make sense to me though i am not familiar with it. can you describe what exactly passthru is or refer me to the docs? will be happy to make this change

Reply:
It's for values that get added to the built derivation (e.g. so someone could do `fxload.libusb` elsewhere to get the value) and it does not get passed to the builder as it would here. I imagine it should be either a passthru attr or in a `let ... in` statement since there's no reason to pass it to the builder, but I'll defer to someone more authoritative if anyone else chimes in saying otherwise

Passthru in docs: https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-passthru

Reply:
i like it! makes sense, i’ll wait as well for more input as i still don’t fully grasp it. i’ll try to use it later today and if i get it to work with passthru will push this change

Reply:
@lilyinstarlight made the change…quite a weird syntax to be honest, i’m still not 100% sure how it works. seems `passthru` object have specific handling. quite anti pattern in functional programming

Reply:
@AndersonTorres what do you think?

Reply:
![E278D92A-F51D-4C53-9653-0385D7DA1E07](https://user-images.githubusercontent.com/1440852/210093754-0259a4aa-8668-40ce-936a-ed1079596fcc.jpeg)

LOL

Reply:
> @lilyinstarlight made the change…quite a weird syntax to be honest, i’m still not 100% sure how it works. seems `passthru` object have specific handling. quite anti pattern in functional programming

It's actually a shorthand for `passthru = { libusb = [...]; }`. There's no special handling, just a bit of a wacky syntax

Reply:
yes, i understood the code will create 
``` 
passthru {
   libusb { 
     withExamples = true;
   }
}
```
which means somewhere there is a code looking for the `passthru` object… rather it being a declared function . anyway, probably belong to a discussion somewhere on nix language

Reply:
Ah that's how you meant! Yeah `passthru` is an argument to `stdenv.mkDerivation` that gets merged to the derivation attrset after `mkDerivation` processing (see [`pkgs/stdenv/generic/make-derivation.nix`](https://github.com/NixOS/nixpkgs/blob/171ac46b9f63ec61166307af9f4aff03377f88a6/pkgs/stdenv/generic/make-derivation.nix#L507-L511))

So the "declared function" you're thinking about is probably `stdenv.mkDerivation` and it would pass in:
```
stdenv.mkDerivation {
  [...];
  passthru = {
    libusb = [libusbDerviationWithExamples];
  };
  [...];
}
```
and `[libusbDerivationWithExamples]` would be accessible as `fxload.libusb` elsewhere in nixpkgs

But yeah `mkDerivation` semantics is not really a discussion for here

Feel free to use a `let ... in` statement at the top of the file instead of `passthru` if you'd rather. Or @AndersonTorres can say it's fine as a builder argument too as you originally wrote. I'm really not invested in either way -- it was just a question of whether it should be changed to match style